File: cross-application-feature-development-management/Applications/IdeManagement/IdeManagement.cs
using cross_application_feature_development_management.Interfaces;

namespace cross_application_feature_development_management.Applications.IdeManagement
{
    public class IdeManagement(
        ICommandLineArgs commandLineArgs,
        IProcessManager processManager,
        ICloseProcessManagement closeProcessManagement,
        IIdeProcessManagement ideProcessManagement
    ): IIdeManagement
    {
        public string GetCommand()
        {
            var command = commandLineArgs.GetByKey("--command");
            return command;
        }

        private bool IsOpen()
        {
            return GetCommand() == "open";
        }

        private bool IsClose()
        {
            return GetCommand() == "close";
        }

        public void Run()
        {
            if (IsOpen())
            {
                ideProcessManagement.Open();
            }
            else if (IsClose())
            {
                ideProcessManagement.Close();
            }
        }
    }

    public interface IIdeManagement
    {
        public void Run();
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/IdeManagement/IdeManagement.fs
namespace cafdemwimu.console.Applications.IdeManagement

open cafdemwimu.console

type IdeManagement(ideProcessManagement: IdeProcessManagement) =
    member private _.GetCommand() =
        let command = CommandLineArgs.GetByKey("--command")
        command

    member private this.IsOpen() = this.GetCommand() = "open"

    member private this.IsClose() = this.GetCommand() = "close"

    member this.Run() =
        if this.IsOpen() then
            ideProcessManagement.Open()
        elif this.IsClose() then
            ideProcessManagement.Close()
